pyccel/parser/semantic.py

```diff
@@ -255,10 +255,12 @@ def annotate(self, **settings):
         return ast
 
     def _get_new_name(self, current_name):
+        if current_name not in self._possible_names:
+            return current_name
+        current_name = current_name + '_'
```

Comment:
It looks like in most cases this function will return the name above, which ends with an underscore.
Is this what we really want?

Reply:
In most cases it won't reach the part where the underscore is added. If it reaches it then it definitely enters the while loop

Reply:
If `current_name` is not among the `_possible_names` then you add an underscore to it. After that change it will not be in `_possible_names` anymore, therefore the interpreter will not enter the loop...

Reply:
Ah! You're right sorry. I wanted a do while but this is python ^^